tests/test_utils.py

```diff
@@ -74,3 +74,28 @@ def test_try_value_to_bool_not_strict(self):
         unchanged_values = ('/path/to/file', 'YES!', '/tmp/certificate')
         for v in unchanged_values:
             self.assertEqual(v, utils.try_value_to_bool(v, strict_mode=False))
+
+    def test_generate_duid(self):
+        # DUID mac address starts from position 12
+        duid_mac_start_point = 12
+
+        duid_count = 10
+        mac = 'fa:16:3e:bd:ce:14'
+        duids = [utils.generate_duid(mac) for _ in range(duid_count)]
```

Comment:
This test (and one below), which iterates generate_duid() is not valid any more.

I think there should at least be one test that validate the format of the generated duid, including the expectation that the 2-4 octets are not all 00.